### PR TITLE
Philipp/perf testing

### PR DESCRIPTION
**Pull request by dasteihn**
Hey hgn,

please pull this. I applied the improvements you requested and added a constant for cleanliness.

On my x86-machine, this program now works with SCHED_DEADLINE and being set to Core #0. Testing with perf also works.

### CODE REVIEW

File: configure_cpu-set.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+mount -t cgroup -o cpuset cpuset /dev/cpuset
+cd /dev/cpuset
+mkdir cpu0
+echo 0 > cpu0/cpuset.cpus
+echo 0 > cpu0/cpuset.mems
+echo 1 > cpuset.cpu_exclusive
+echo 0 > cpuset.sched_load_balance
+echo 1 > cpu0/cpuset.cpu_exclusive
+echo 1 > cpu0/cpuset.mem_exclusive
+echo $$ > cpu0/tasks
```
Comment (hgn) [inline configure_cpu-set.sh]: no mix of underscores and dash - terrible.

Please move this file into analyze/ directory.

Please source the file in each analyzer shell script.

File: src/runner.c [hunk 2]
```diff
@@ -109,13 +111,13 @@ void parse_args(struct config *cfg, int argc, char *argv[])
 			break;
 		/* scheduler times are ns, user gives ms */
 		case 'r':
-			cfg->attr.sched_runtime = 1000*1000*atoi(optarg);
+			cfg->attr.sched_runtime = MILLION * atoi(optarg);
```
Comment (hgn) [inline src/runner.c]: this variable should not be called "million", it should be called NANOSECONDS or something else. That what you want to do

File: configure_cpu-set.sh (hunk 1 above)
Comment (dasteihn) [reply configure_cpu-set.sh]: What do you mean? I am afraid there is no dash. Alter.
The names are given by the kernel, not me

File: src/runner.c (hunk 2 above)
Comment (dasteihn) [reply src/runner.c]: I will try to make up something smart

File: configure_cpu-set.sh (hunk 1 above)
Comment (hgn) [reply configure_cpu-set.sh]: configure_cpu-set.sh -> configure-cpu-set.sh

File: src/runner.c (hunk 2 above)
Comment (dasteihn) [reply src/runner.c]: I do not think there is something smarter than million (except maybe E6 etc.)

The thing is: We want the user to be able to hand over time parameters, as milliseconds, whereas the kernel interprets these integers as nanoseconds. Therefore, we have to multiply by one million. The only other option I see is calling it something like `MS_TO_NS_FACTOR` or something akin

File: configure_cpu-set.sh (hunk 1 above)
Comment (dasteihn) [reply configure_cpu-set.sh]: I did this intentionally to show which words belong to each other. Will change it if you want

**Comment by dasteihn** [comment]
btw. there are some minor issues I'm having.

1.  In your shell analyzation scripts you define `sleep X` to control how long our test is running. In my eyes this is colliding with parameter `-T` which we use to tell the runner how many program iterations it has to complete until ending itself. We do this to avoid freezing the CPU. Apparently your shell script is doing the same, so we have two things being responsible for the same job

2. We configure a parameter telling the program how long it has to sleep until doing busy-loops again. I am not sure how useful having an exact parameter is here. I already know that with sched_deadline, the scheduler will win. Meaning: If your period is 1000 ms, you calculate 100ms and then try to sleep 200ms, you will not wake up after 300ms, but after 1000ms.

Alter.

**Comment by hgn** [comment]
Right, MS_TO_NS_FACTOR is what I am looking for. Am 06.12.2019 21:44 schrieb dasteihn <notifications@github.com>:@dasteihn commented on this pull request.



In src/runner.c:
> @@ -109,13 +111,13 @@ void parse_args(struct config *cfg, int argc, char *argv[])
 			break;
 		/* scheduler times are ns, user gives ms */
 		case 'r':
-			cfg->attr.sched_runtime = 1000*1000*atoi(optarg);
+			cfg->attr.sched_runtime = MILLION * atoi(optarg);

I do not think there is something smarter than million (except maybe E6 etc.)
The thing is: We want the user to be able to hand over time parameters, as milliseconds, whereas the kernel interprets these integers as nanoseconds. Therefore, we have to multiply by one million. The only other option I see is calling it something like MS_TO_NS_FACTOR or something akin

**Comment by hgn** [comment]
ACKAm 06.12.2019 21:44 schrieb dasteihn <notifications@github.com>:@dasteihn commented on this pull request.



In configure_cpu-set.sh:
> @@ -0,0 +1,13 @@
+#/bin/bash
+
+mkdir /dev/cpuset
+mount -t cgroup -o cpuset cpuset /dev/cpuset
+cd /dev/cpuset
+mkdir cpu0
+echo 0 > cpu0/cpuset.cpus
+echo 0 > cpu0/cpuset.mems
+echo 1 > cpuset.cpu_exclusive
+echo 0 > cpuset.sched_load_balance
+echo 1 > cpu0/cpuset.cpu_exclusive
+echo 1 > cpu0/cpuset.mem_exclusive
+echo $$ > cpu0/tasks

I did this intentionally to show which words belong to each other. Will change it if you want

**Pull request merged**